src/marks/raster.js

```diff
@@ -282,30 +282,14 @@ export function interpolateNone(index, width, height, X, Y, V) {
 
 export function interpolatorBarycentric({random = randomLcg(42)} = {}) {
   return (index, width, height, X, Y, V) => {
-    // Flatten the input coordinates to prepare to insert extrapolated points
-    // along the perimeter of the grid (so there’s no blank output).
-    const n = index.length;
-    const nw = width >> 2;
-    const nh = (height >> 2) - 1;
-    const m = n + nw * 2 + nh * 2;
-    const XY = new Float64Array(m * 2);
-    for (let i = 0; i < n; ++i) (XY[i * 2] = X[index[i]]), (XY[i * 2 + 1] = Y[index[i]]);
-
-    // Add points along each edge, making sure to include the four corners for
-    // complete coverage (with no chamfered edges).
-    let i = n;
-    const addPoint = (x, y) => ((XY[i * 2] = x), (XY[i * 2 + 1] = y), i++);
-    for (let j = 0; j <= nw; ++j) addPoint((j / nw) * width, 0), addPoint((j / nw) * width, height);
-    for (let j = 0; j < nh; ++j) addPoint(width, (j / nh) * height), addPoint(0, (j / nh) * height);
-
-    // To each edge point, assign the closest (non-extrapolated) value.
-    V = take(V, index);
-    const delaunay = new Delaunay(XY.subarray(0, n * 2));
-    for (let j = n, ij; j < m; ++j) V[j] = V[(ij = delaunay.find(XY[j * 2], XY[j * 2 + 1], ij))];
-
     // Interpolate the interior of all triangles with barycentric coordinates
-    const {points, triangles} = new Delaunay(XY);
+    const {points, triangles, hull} = Delaunay.from(
+      index,
+      (i) => X[i],
+      (i) => Y[i]
+    );
     const W = new V.constructor(width * height);
+    const I = new Uint8Array(width * height);
```

Comment:
Having both `index` and `I` here is hard to follow, since both are indexes. Maybe `I` should be renamed to `delaunayIndex` or something? I’m not entirely sure that the right index is being used in the right place… I wonder if we are assuming that `index` is [0, 1, 2, 3, …]. Worth double-checking if you haven’t already.

Reply:
I've renamed I into S and added a comment.

The values are extracted here from the indexed data—I've added a test to triple-check.
```
  X = Float64Array.from(hull, (i) => X[index[i]]);
  Y = Float64Array.from(hull, (i) => Y[index[i]]);
  V = Array.from(hull, (i) => V[index[i]]);
```